internal/command/stacks.go

```diff
@@ -0,0 +1,363 @@
+
+	bm, err := pluginshared.NewStacksBinaryManager(ctx, packagesPath, overridePath, c.pluginService, runtime.GOOS, runtime.GOARCH)
+	if err != nil {
+		return diags.Append(tfdiags.Sourceless(tfdiags.Error, errorSummary, err.Error()))
+	}
+
+	version, err := bm.Resolve()
+	if err != nil {
+		return diags.Append(tfdiags.Sourceless(tfdiags.Error, "Stacks plugin download error", err.Error()))
+	}
+
+	var cacheTraceMsg = ""
+	if version.ResolvedFromCache {
+		cacheTraceMsg = " (resolved from cache)"
+	}
+	if version.ResolvedFromDevOverride {
+		cacheTraceMsg = " (resolved from dev override)"
+		detailMsg := fmt.Sprintf("Instead of using the current released version, Terraform is loading the stacks plugin from the following location:\n\n - %s\n\nOverriding the stacks plugin location can cause unexpected behavior, and is only intended for use when developing new versions of the plugin.", version.Path)
+		diags = diags.Append(tfdiags.Sourceless(
+			tfdiags.Warning,
+			"Stacks plugin development overrides are in effect",
+			detailMsg,
+		))
+	}
+	log.Printf("[TRACE] plugin %q binary located at %q%s", version.ProductVersion, version.Path, cacheTraceMsg)
+	c.pluginBinary = version.Path
+	return diags
+}
+
+func (c *StacksCommand) initPackagesCache() (string, error) {
+	packagesPath := path.Join(c.WorkingDir.DataDir(), StacksPluginDataDir)
+
+	if info, err := os.Stat(packagesPath); err != nil || !info.IsDir() {
+		log.Printf("[TRACE] initialized stacksplugin cache directory at %q", packagesPath)
+		err = os.MkdirAll(packagesPath, 0755)
+		if err != nil {
+			return "", fmt.Errorf("failed to initialize stacksplugin cache directory: %w", err)
+		}
+	} else {
+		log.Printf("[TRACE] stacksplugin cache directory found at %q", packagesPath)
+	}
+
+	return packagesPath, nil
+}
+
+// Run runs the stacks command with the given arguments.
+func (c *StacksCommand) Run(args []string) int {
+	args = c.Meta.process(args)
+	return c.realRun(args, c.Meta.Streams.Stdout.File, c.Meta.Streams.Stderr.File)
+}
+
+// Help returns help text for the stacks command.
+func (c *StacksCommand) Help() string {
+	helpText := new(bytes.Buffer)
+	if exitCode := c.realRun([]string{}, helpText, io.Discard); exitCode != 0 {
+		return ""
+	}
+
+	return helpText.String()
+}
+
+// Synopsis returns a short summary of the stacks command.
+func (c *StacksCommand) Synopsis() string {
+	return "Manage HCP Terraform stack operations"
+}
+
+// StacksPluginConfig is everything the stacks plugin needs to know to configure a
+// client and talk to HCP Terraform.
+type StacksPluginConfig struct {
+	Address             string `md:"tfc-address"`
+	BasePath            string `md:"tfc-base-path"`
+	DisplayHostname     string `md:"tfc-display-hostname"`
+	Token               string `md:"tfc-token"`
+	TerraformBinaryPath string `md:"terraform-binary-path"`
+	OrganizationName    string `md:"tfc-organization"`
+	ProjectName         string `md:"tfc-project"`
+	StackName           string `md:"tfc-stack"`
+}
```

Comment:
One small thing we're missing here is the capturing of terminal width. I _believe_ it is accessible through the command's `Meta.Streams` 🤞.